testing/trino-product-tests/src/main/java/io/trino/tests/product/jdbc/TestKerberosConstrainedDelegationJdbc.java

```diff
@@ -0,0 +1,126 @@
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.Properties;
+
+import static com.google.common.base.Throwables.throwIfInstanceOf;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryResult.forResultSet;
+import static io.trino.tests.product.TestGroups.JDBC_KERBEROS_CONSTRAINED_DELEGATION;
+import static java.lang.String.format;
+import static org.ietf.jgss.GSSCredential.DEFAULT_LIFETIME;
+import static org.ietf.jgss.GSSCredential.INITIATE_ONLY;
+import static org.ietf.jgss.GSSName.NT_USER_NAME;
+
+public class TestKerberosConstrainedDelegationJdbc
+        extends ProductTest
+{
+    private static final String KERBEROS_OID = "1.2.840.113554.1.2.2";
+    @Inject
+    @Named("databases.presto.jdbc_url")
+    String jdbcUrl;
+
+    @Inject
+    @Named("databases.presto.kerberos_principal")
+    private String kerberosPrincipal;
+
+    @Inject
+    @Named("databases.presto.kerberos_keytab")
+    private String kerberosKeytab;
+
+    private GSSManager gssManager;
+    private KerberosAuthentication kerberosAuthentication;
+
+    @BeforeMethodWithContext
+    public void setUp()
+    {
+        this.gssManager = GSSManager.getInstance();
+        this.kerberosAuthentication = new KerberosAuthentication(kerberosPrincipal, kerberosKeytab);
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testSelectConstrainedDelegationKerberos()
+            throws Exception
+    {
+        Properties driverProperties = new Properties();
+        GSSCredential credential = createGssCredential();
+        driverProperties.put("KerberosConstrainedDelegation", credential);
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, driverProperties);
+                PreparedStatement statement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThat(forResultSet(results)).matches(TpchTableResults.PRESTO_NATION_RESULT);
+        }
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testCtasConstrainedDelegationKerberos()
+            throws Exception
+    {
+        Properties driverProperties = new Properties();
+        GSSCredential credential = createGssCredential();
+        driverProperties.put("KerberosConstrainedDelegation", credential);
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, driverProperties);
+                PreparedStatement statement = connection.prepareStatement(format("CREATE TABLE %s AS SELECT * FROM tpch.tiny.nation", "test_kerberos_ctas"))) {
+            int results = statement.executeUpdate();
+            Assertions.assertThat(results).isEqualTo(25);
+        }
+    }
+
+    private GSSCredential createGssCredential()
+            throws GSSException
```

Comment:
nit: We can directly throw an Exception.

Reply:
refactored